p2p/libp2p/identity/identity.go

```diff
@@ -0,0 +1,125 @@
+package identity
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+
+	"github.com/pokt-network/pocket/p2p/common"
+	"github.com/pokt-network/pocket/p2p/transport"
+	"github.com/pokt-network/pocket/p2p/types"
+	poktCrypto "github.com/pokt-network/pocket/shared/crypto"
+)
+
+var (
+	ErrIdentity = common.NewErrFactory("")
+)
+
+// PoktPeerFromStream builds a pokt peer from a libp2p stream.
+// (NOTE: excludes `ServiceURL`)
+func PoktPeerFromStream(stream network.Stream) (*types.NetworkPeer, error) {
+	remotePubKeyBytes, err := stream.Conn().RemotePublicKey().Raw()
+	// NB: abort handling this stream.
+	if err != nil {
+		return nil, err
+	}
+	poktPubKey, err := poktCrypto.NewPublicKeyFromBytes(remotePubKeyBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.NetworkPeer{
+		Dialer:    transport.NewLibP2PTransport(stream),
+		PublicKey: poktPubKey,
+		// NB: pokt analogue of libp2p peer.ID
```

Comment:
Maybe call this `TECHDEBT`? In our `Makefile we have a bunch of things you can search for (free free to update)

<img width="836" alt="Screenshot 2023-02-13 at 8 31 55 PM" src="https://user-images.githubusercontent.com/1892194/218639987-bc0714cc-e0d6-4a9f-bb79-4aff8f2e41ca.png">

Reply:
Thanks for sharing that, I was not previously aware! :star_struck: 

My intention here was to point out that category theoretical equality of the two concepts. 

I'm not sure I see how it's `TECHDEBT`, are you trying to call out volatility due to the forthcoming consolidation of identity?